x/emissions/keeper/keeper.go

```diff
@@ -1864,43 +1864,48 @@ func (k *Keeper) removeFromTopicMempool(ctx context.Context, topicId TopicId, re
 
 // Throws an error if we are already at capacity
 // Does not check for pre-existing inclusion in the topic mempool before adding!
-func (k *Keeper) AddToMempool(ctx context.Context, request types.InferenceRequest) error {
+func (k *Keeper) AddToMempool(ctx context.Context, request types.InferenceRequest) (Uint, error) {
 	hasRequest, err := k.requests.Has(ctx, request.Id)
 	if err != nil {
-		return err
+		return cosmosMath.Uint{}, err
 	}
 	if hasRequest {
-		return types.ErrRequestAlreadyExists
+		return cosmosMath.Uint{}, types.ErrRequestAlreadyExists
 	}
 
 	count, err := k.GetTopicMempoolRequestCount(ctx, request.TopicId)
 	if err != nil {
-		return err
+		return cosmosMath.Uint{}, err
 	}
 	maxCount, err := k.GetParamsMaxRequestsPerTopic(ctx)
 	if err != nil {
-		return err
+		return cosmosMath.Uint{}, err
 	}
 	if count >= maxCount {
-		return types.ErrTopicMempoolAtCapacity
+		return cosmosMath.Uint{}, types.ErrTopicMempoolAtCapacity
 	}
 
 	err = k.requests.Set(ctx, request.Id, request)
 	if err != nil {
-		return err
+		return cosmosMath.Uint{}, err
 	}
 
 	newIndex, err := k.IncrementTopicMempoolRequestCount(ctx, request.TopicId)
 	if err != nil {
-		return err
+		return cosmosMath.Uint{}, err
 	}
 
 	err = k.addToTopicMempool(ctx, request.TopicId, newIndex-1, request.Id)
 	if err != nil {
-		return err
+		return cosmosMath.Uint{}, err
+	}
+
+	err = k.AddUnmetDemand(ctx, request.TopicId, request.BidAmount)
+	if err != nil {
+		return cosmosMath.Uint{}, err
 	}
 
-	return k.AddUnmetDemand(ctx, request.TopicId, request.BidAmount)
+	return k.GetTopicUnmetDemand(ctx, request.TopicId)
```

Comment:
`AddUnmetDemand` already get the topic unmet demand, you can just return it instead of querying again

Reply:
Okay